src/scheduler/check.cpp

```diff
@@ -1571,26 +1571,44 @@ check_normal_node_path(status *policy, server_info *sinfo, queue_info *qinfo, re
 			if (qinfo->has_nodes)
 				ninfo_arr = qinfo->nodes;
 			else if (!(sinfo->svr_to_psets.empty())) {	/* Multi-server psets */
+				int i = 0;
+				int owner_set = -1;
+
 				/* Find the pset of the server which owns the job */
 				for (auto &spset: sinfo->svr_to_psets) {
 					if (spset.svr_inst_id == resresv->job->svr_inst_id) {
 						msvr_pset[0] = spset.np;
-						if (!resresv->job->is_array && spec->total_chunks == 1) {
-							/* Allow job to be placed on nodes of other servers as well
-							 * For now, restricting job arrays and multi-chunk jobs
-							 * to nodes of just the local server
-							 */
-							msvr_pset[1] = sinfo->allpart;
-							msvr_pset[2] = NULL;
+						owner_set = i;
+						break;
+					}
+					i++;
+				}
+				if (owner_set != -1) {
+					if (resresv->job->is_array)	/* Restrict job arrays to owner server */
+						msvr_pset[1] = NULL;
+					else if (spec->total_chunks == 1) {	/* Single chunk jobs */
```

Comment:
I think we want to restrict the else if condition only to jobs right ? Otherwise, it allows reservations to span across server instances in case if the specification contains a single chunk.